### ISSUE TITLE
Finalize variable names for arguments returned by parse

### ISSUE THREAD

**Issue by shadowspawn** [pkgjs/parseargs#12]
I am unsure what semantic meaning to give `args` as a name in the returned object. To be fair, this is made worse by my experience with Commander, where `program.args` are the "positionals" left after parsing by removing the options and option-values.

I suspect it is short for "arguments" and skirting around what terminology to use. I think is reasonable to describe the input parameters to `parseArgs` as "arguments", but less clear when talking about the results of parsing. 

The Initial Proposal returned `options` and `positionals`, and raised the question of terminology (and used "Flags", "Options", "Positionals"):
 - questions: https://github.com/nodejs/node/pull/35015#issue-690492606
 - feedback query: https://github.com/nodejs/node/pull/35015#issuecomment-685911644

**Comment by bcoe** [comment]
@shadowspawn I kind of like `values` and `flags`, thoughts?

**Comment by shadowspawn** [comment]
Did a bit of a survey. Terminology used in programmer guides to describe options/flags:

- [Commander](https://github.com/tj/commander.js/blob/master/docs/terminology.md): option
- [Yargs](https://github.com/yargs/yargs/blob/main/docs/api.md): option
- [minimist](https://github.com/substack/minimist): option

- [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html): option
- [argparse](https://docs.python.org/3.10/library/argparse.html): optional argument
- [oclif](https://oclif.io/docs/flags) command flag
- [Cobra](https://github.com/spf13/cobra#concepts) flag
- man pages usually have an OPTIONS section

Programming object holding options/flags after parsing:

- Commander: object of options (key:value)
- Yargs: object of options (key:value, and positionals in `_`)
- minimist: : object of options (key:value, and positionals in `_`)

- argparse: object with combined option and arguments (key:value)
- oclif: result object has `flags` property, object of options (key:value)

**Comment by shadowspawn** [comment]
> `values` and` flags`

Short version: better, yes.

**Comment by ljharb** [comment]
To me, flags are only booleans; “option” seems like the most universal and intuitive term. An argument can be an option argument, or a positional argument; an option can be a flag, or not.

**Comment by shadowspawn** [comment]
Long comment, not disagreeing.

I see the term "flag" get used three ways:
- "flag" rather than "option" to describe the whole feature of non-positional (e.g. Cobra and Oclif)
- "flag" to describe options which do not take a value (as ljharb describes)
- "flag" to describe the argument with the hyphen or double hyphen in front of if, and hence short-flag and long-flag

Personally, I was using both the second and third forms in documentation. When I noticed my own inconsistency, I switched to saying "boolean option" for an option not taking an option-argument so using "flag" just one way.

I am still pretty comfortable with "flag" as term for not taking an option-argument. (And staying away from mentioning boolean might decrease assumptions leading to the `--explode=false` problem. 😄 )
